data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/RecordType.java

```diff
@@ -15,15 +17,27 @@
  * TODO: remove in 2.0
  */
 enum RecordType {
-    otlp,
-    event;
+    @JsonProperty("otlp")
```

Comment:
Does this usage of `@JsonProperty` work? I think this defines a property with a key name of `otlp`.

I do know that using `@JsonCreator` and a static factory method will work.

Reply:
Actually I am not able to get @JsonCreator work, but this works as long as the recordType in OTelTraceSourceConfig is annotated with @JsonProperty.

Reply:
Oh. I just noticed this in the documentation:

> Starting with Jackson 2.6 this annotation may also be used to change serialization of Enum like so:

Nice!